### PR TITLE
Fix `Result<SyncReturn(T), E>`

### PR DESCRIPTION
**Pull request by rogurotus**
Fix `Result<SyncReturn(T), E>`

Now it always returns "Box\<intoDart\>"

### CODE REVIEW

**Comment by rogurotus** [comment]
ping, @fzyzcjy

**Comment by fzyzcjy** [comment]
Sure! I will review ASAP (possibly today)

Btw I did not receive notifications when it is marked ready for review (or did I miss it) :/

File: frb_dart/lib/src/basic.dart [hunk 1]
```diff
@@ -88,11 +88,12 @@ abstract class FlutterRustBridgeBase<T extends FlutterRustBridgeWireBase> {
       if (isSuccess) {
         return task.parseSuccessData(rawReturn);
       } else {
-        throw FfiException('EXECUTE_SYNC', rawReturn as String, null);
+        if (rawReturn is String) {
+          throw FfiException('EXECUTE_SYNC', rawReturn, null);
+        } else {
+          throw task.parseErrorData!(rawReturn);
+        }
```
Comment (fzyzcjy) [inline frb_dart/lib/src/basic.dart:95]: Not sure whether the `String` case is indeed a panic? If so, maybe we have https://github.com/fzyzcjy/flutter_rust_bridge/blob/392a87f07d49e5df584633b5811a0eefa82da3ae/frb_dart/lib/src/basic.dart#L229

File: frb_rust/src/handler.rs [hunk 2]
```diff
@@ -321,7 +321,10 @@ impl ErrorHandler for ReportDartErrorHandler {
     }
 
     fn handle_error_sync(&self, error: Error) -> WireSyncReturn {
-        wire_sync_from_data(error.message(), false)
+        match error {
+            Error::CustomError(err) => wire_sync_from_data(err.box_into_dart(), false),
+            Error::Panic(err) => wire_sync_from_data(error_to_string(&err), false),
+        }
```
Comment (fzyzcjy) [inline frb_rust/src/handler.rs:327]:
```suggestion
let data =         match error {
            Error::CustomError(err) =>err.box_into_dart()
            Error::Panic(err) =>error_to_string(&err)
        };
        wire_sync_from_data(data, false)
```

**Review by fzyzcjy — APPROVED** [review]
Only nits and ready to merge!

File: frb_rust/src/handler.rs (hunk 2 above)
Comment (rogurotus) [reply frb_rust/src/handler.rs:327]: it won't work, there are different types
CustomError - wire_sync_from_data<DartAbi>
Panic - wire_sync_from_data<String>

Comment (rogurotus) [reply frb_rust/src/handler.rs:327]: We can `wire_sync_from_data(error.into_dart(), false)`

File: frb_dart/lib/src/basic.dart (hunk 1 above)
Comment (rogurotus) [reply frb_dart/lib/src/basic.dart]: refacted

File: frb_example/pure_dart/dart/lib/main.dart [hunk 3]
```diff
@@ -243,7 +243,12 @@ void main(List<String> args) async {
       try {
         api.handleSyncReturn(mode: mode);
         fail("exception not thrown");
-      } on FfiException catch (e) {
+      } catch (e) {
+        if (mode == 'RESULT_ERR') {
+          e as FrbAnyhowException;
+        } else {
+          e as PanicException;
```
Comment (fzyzcjy) [inline frb_example/pure_dart/dart/lib/main.dart:250]: nit: sth like `expect(e, isA<WhateverType>());`

**Review by fzyzcjy — APPROVED** [review]

**Comment by fzyzcjy** [comment]
Good refactor! The old integer-based solution was ugly (yes it was written by myself at the first few days of the repository when I did not expect it to grow popular ;) )

File: frb_example/pure_dart/dart/lib/main.dart (hunk 3 above)
Comment (rogurotus) [reply frb_example/pure_dart/dart/lib/main.dart]: fix

**Comment by fzyzcjy** [comment]
Great job!

**Pull request merged**